### PR TITLE
Add contract error handler

### PR DESCRIPTION
**Pull request by gyroflaw**
Fix https://github.com/casper-ecosystem/erc20/issues/61

### CODE REVIEW

**Review by hoffmannjan — CHANGES_REQUESTED** [review]
Overall looks good, just a few changes to stay consistent with CEP-78 ☺️ Good job!

File: client-js/src/ERC20Client.ts [hunk 1]
```diff
@@ -274,4 +277,67 @@ export default class ERC20Client {
...
+            contractErrorMessagePrefix.length,
+            error_message.length
+          ),
+          10
+        );
+        throw new ContractError(errorCode);
+      } else throw new Error(error_message);
+    }
+    return result;
+  }
```
Comment (hoffmannjan) [inline client-js/src/ERC20Client.ts:342]: maybe we should have a method like this in `casper-js-sdk` ? I wouldn't add this to contract client :) Especially that it is used only in tests

File: client-js/src/error.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  ERROR_INSUFFICIENT_BALANCE = MAX_U16 - 1,
+  ERROR_INSUFFICIENT_ALLOWANCE = MAX_U16 - 2,
+  ERROR_OVERFLOW = MAX_U16 - 3
+}
+
+export class ContractError extends Error {
+  constructor(public code: number) {
+    super(ERROR_CODES[code]);
+  }
+}
```
Comment (hoffmannjan) [inline client-js/src/error.ts:14]: super nice

File: client-js/README.md [hunk 3]
```diff
@@ -38,9 +41,9 @@ npm i casper-erc20-js-client
     [owner]
   );
 
-  await client.putDeploy(deploy);
+  const deployHash = await erc20.putDeploy(deploy);
```
Comment (hoffmannjan) [inline client-js/README.md:44]: this is not actual anymore, right?

File: client-js/README.md [hunk 4]
```diff
@@ -38,9 +41,9 @@ npm i casper-erc20-js-client
...
 
-  const result = await client.nodeClient.waitForDeploy(deploy);
+  const result = await erc20.waitForDeploy(deployHash);
```
Comment (hoffmannjan) [inline client-js/README.md:46]: please remove this part, I think this is responsibility of the developer to handle this - we don't need provide method for this.

**Review by hoffmannjan — CHANGES_REQUESTED** [review]
Few last nitpicks and we can merge

File: client-js/src/ERC20Client.ts (hunk 1 above)
Comment (gyroflaw) [reply client-js/src/ERC20Client.ts:342]: the purpose of this method was to parse contract error, if you don't like it approach then we can create a method like `parseDeployResult`

Comment (hoffmannjan) [reply client-js/src/ERC20Client.ts:342]: yeah I thought about it, it will be more universal for end users :)

**Review by hoffmannjan — APPROVED** [review]

**Pull request merged**